src/init/caches.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INIT_CACHES_H
+#define BITCOIN_INIT_CACHES_H
+
+#include <cstddef> // for size_t
+#include <cstdint> // for int64_t
+
+class ArgsManager;
+
+struct CacheSizes {
+    int64_t block_tree_db_cache_size;
+    int64_t coin_db_cache_size;
+    int64_t coin_cache_usage_size;
+    int64_t tx_index_cache_size;
+    int64_t filter_index_cache_size;
+};
+void CalculateCacheSizes(const ArgsManager& args, CacheSizes& cache_sizes, size_t n_indexes = 0);
```

Comment:
Again, it'd be nice to have a doxygen comment here. It's not immediately obvious what `n_indexes` represents.

Is there any reason that you made `cache_sizes` an out-param rather than the return value for this function?

Since this is a new function and you're adding all the call sites, can we make `n_indexes` a required argument?

Reply:
> Is there any reason that you made cache_sizes an out-param rather than the return value for this function?

https://github.com/bitcoin/bitcoin/pull/23280#discussion_r739436153

Reply:
In my latest push (6736e24a490d756e0e1e70b9e212775cf305ebe6), I ended up having `CalculateCacheSizes` return a CacheSize, which is perhaps a bit more readable.